src/main/java/org/embulk/input/jira/Issue.java

```diff
@@ -0,0 +1,152 @@
+package org.embulk.input.jira;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonNull;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.StreamSupport;
+
+public class Issue
+{
+    private JsonObject flatten;
+    private JsonObject json;
+
+    public Issue(JsonObject original)
+    {
+        this.json = original;
+    }
+
+    public JsonElement getValue(String path)
+    {
+        List<String> keys = new ArrayList<>(Arrays.asList(path.split("\\.")));
+        return get(json, keys);
+    }
+
+    private JsonElement get(JsonElement json, List<String> keys)
+    {
+        if (json == null || json.isJsonNull()) {
+            return JsonNull.INSTANCE;
+        }
+        else if (keys.isEmpty() || (json.isJsonArray() && json.getAsJsonArray().size() == 0)) {
+            return json;
+        }
+        String key = keys.get(0);
+        keys.remove(0);
+        if (json.isJsonArray()) {
+            JsonArray arrays = new JsonArray();
+            for (JsonElement elem : json.getAsJsonArray()) {
+                if (elem.isJsonObject()) {
+                    arrays.add(elem.getAsJsonObject().get(key));
+                }
+                else {
+                    arrays.add(elem);
+                }
+            }
+            return get(arrays, keys);
+        }
+        else {
+            return get(json.getAsJsonObject().get(key), keys);
+        }
+    }
+
+    public void toRecord()
```

Comment:
These mutable, stateful methods usually prompt for errors. Is there anything preventing us from compute `flatten` when calling `getFlatten()` (and certainly cache it) instead of having to remember to call `toRecord` before that?

Reply:
Should we so defensive here ?
In the beginning of my development, I put it in the constructor. 
But after that I realize this method only trigger on guessing. And I can freely control whenever to call it. So I decided to put it as it is.

Reply:
It's just that the current implementation has room for error, while the alternative one with pretty much same number of lines of code doesn't. But it doesn't mean your current one is incorrect and the decision is yours, I consider this is resolved 👍